src/main/kotlin/com/michalpu/PokeRest/pokemon/TypeRelations.kt

```diff
@@ -0,0 +1,68 @@
+package com.michalpu.PokeRest.pokemon
+
+class TypeRelations private constructor(val noDamageTo: Set<String>?,
```

Comment:
`PokemonTypeRelation`
I would use another type than String or include the meaning of this type in a variable name. What are the strings inside the set? Ids, names or something else? 

you wrap in into another class like 
```
class PokemonRelation(
   val name: String
)
```
and then use `noDamageTo: Set<PokemonRelation>` which will indicate immediately what's the meaning of this variable.